src/main.rs

```diff
@@ -395,6 +402,12 @@ async fn main_impl() -> Result<()> {
             print_add_response(hash, entries);
             Ok(())
         }
+        Commands::Addresses { addr } => {
+            let listen_addr = addr.unwrap_or_else(|| DEFAULT_BIND_ADDR.parse().unwrap());
```

Comment:
Hmmmm, so initially I went with that approach, but that didn't yield results till I pulled in flubs PR (the LocalAddress) thing is still not glued in. Once I did that, I saw that there is generally little point since the process is entirely independent of the bound socket. It's also the same way we inject the IP into the ticket hence, if you give it the same input `addr` it should produce the correct result. Which makes this whole thing a lot simpler and less cruft for a utility func.

The only thing that actually puts me off of this is that we have a top level utility func at all. We had this discussion on the original implementation about avoiding unnecessary CLI cruft.

From a technical perspective I'm fine with either, your call.

Reply:
Well the `local_addr()` on the socket will still be unspecified, but it will contain the correct port, which we don't know in the case where we don't forward this call. 
In general I think we should have a method like `available_addresses()` on the provider, which will do this work, and this cli call just forwards to that.

In terms of cli cruft, I would say this will likely go into some subcommand eventually, but lets keep it here for now, as much as I dislike it, it has it's use.